pkg/multus/multus.go

```diff
@@ -896,5 +897,22 @@ func CmdDel(args *skel.CmdArgs, exec invoke.Exec, kubeClient *k8s.ClientInfo) er
 		}
 	}
 
-	return delPlugins(exec, pod, args, k8sArgs, in.Delegates, len(in.Delegates)-1, in.RuntimeConfig, in)
+	e := delPlugins(exec, pod, args, k8sArgs, in.Delegates, len(in.Delegates)-1, in.RuntimeConfig, in)
+
+	// delegate plugin delete success, delete cache file
+	// others
+	// 1.  sandBox container stop times, but delegated plugin failed (cache file has been deleted)
+	// 2.  container gc manager clean up forever even though POD has been fully deleted from ETCD
+	// this case POD not find && cache file not find; never clean up
+	if removeCacheConf {
+		if in.AllowTryDeleteOnErr {
+			if e == nil {
+				_ = os.Remove(path)
+			}
+		} else {
+			_ = os.Remove(path) // lgtm[go/path-injection]
+		}
+	}
```

Comment:
This conditional could be simplified as follows:

```
        if in.AllowTryDeleteOnErr {
                if removeCacheConf && e == nil {
                        _ = os.Remove(path.Clean(cacheConfPath))
                }
        } else {
                _ = os.Remove(path.Clean(cacheConfPath))
        }
```

Reply:
```
        if in.AllowTryDeleteOnErr {
                if removeCacheConf && e == nil {
                        _ = os.Remove(path)
                }
        } else {
                if removeCacheConf {
                        _ = os.Remove(path)   // path file maybe not existed ,  even though has ignore error ...
                }
        }
```

should like this ?  path file maybe not existed that else ;  add check before remove ?  so I start this conditional with  removeCacheConf  like this   if removeCacheConf {.......}

Reply:
Thank you for the reply. Yeah, my suggested code was wrong and thank you for the pointing it out.

My intention is to 'check condition of `AllowTryDeleteError` first' to clearly show the behavior with/without the option.
So how about to update such like that?

```
        if in.RetryDeleteOnError {
                if removeCacheConf {
                        if e == nil || strings.Contains(e.Error(), "no such file or directory") {
                                _ = os.Remove(path) // lgtm[go/path-injection]
                        }
                }
        } else {
                if removeCacheConf {
                        _ = os.Remove(path) // lgtm[go/path-injection]
                }
        }
```

Reply:
done